src/components/file-dropzone/file-dropzone.tsx

```diff
@@ -0,0 +1,162 @@
+     * Unique file type specifiers can also be used, for example: `.jpg`, `.pdf`.
+     * A comma-separated list of file extensions or MIME types is also acceptable, e.g., `image/png, image/jpeg` or
+     * `.png, .jpg, .jpeg`.
+     *
+     * @see [HTML attribute: accept](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept) for more
+     * details.
+     */
+    @Prop({ reflect: true })
+    public accept: string = '*';
+
+    /**
+     * Set to `true` to disable the file dropzone.
+     */
+    @Prop()
+    public disabled: boolean = false;
+
+    /**
+     * Text to be displayed when the user is dragging a file over the dropzone.
+     */
+    @Prop()
+    public onDragLayoutText: string = '';
+
+    /**
+     * Icon to be displayed when the user is dragging a file over the dropzone.
+     */
+    @Prop()
+    public onDropLayoutIcon: string = 'upload_2';
+
+    /**
+     * Sets to true when there is a file to drop
+     */
+    @State()
+    private hasFileToDrop: boolean = false;
+
+    /**
+     * Emitted when files are selected
+     */
+    @Event()
+    filesSelected: EventEmitter<FileInfo[]>;
+
+    /**
+     * Emitted when files are selected but do not conform with the `accept` property specifications.
+     * This can happen when the file types or formats of the selected files are not among the ones allowed by the dropzone,
+     * as defined by the `accept` property.
+     *
+     * @see {@link accept} for details on how to specify acceptable file types.
+     */
+    @Event()
+    filesRejected: EventEmitter<FileInfo[]>;
+
+    public render() {
+        return (
+            <Host
+                onDrop={this.handleDrop}
+                onDragOver={this.handleDragOver}
+                onDragLeave={this.handleDragLeave}
+            >
+                <slot />
+                {this.renderOnDragLayout()}
+            </Host>
+        );
+    }
+
+    private renderOnDragLayout = () => {
+        if (this.hasFileToDrop) {
+            return (
+                <div class="has-file-to-drop">
+                    <limel-icon
+                        class="dropzone-icon"
+                        name={this.onDropLayoutIcon}
+                    ></limel-icon>
+                    {this.renderOnDragLayoutText()}
+                </div>
+            );
+        }
+    };
+
+    private renderOnDragLayoutText = () => {
+        if (this.onDragLayoutText.length === 0) {
+            return;
+        }
+
+        return <span class="dropzone-text">{this.onDragLayoutText}</span>;
+    };
+
+    private handleDrop = (event: DragEvent) => {
+        event.stopPropagation();
+        event.preventDefault();
+        this.hasFileToDrop = false;
+
+        if (this.disabled) {
+            return;
+        }
+
+        const files: File[] = Array.from(event.dataTransfer.files);
+        const fileInfos: FileInfo[] = files.map(createFileInfo);
+
+        const acceptedFileInfos: FileInfo[] = fileInfos.filter((file) =>
```

Comment:
Maybe try out Lodash's `partition` function for 137-142, it's perfect :)

Reply:
I love it 😍
See https://github.com/Lundalogik/lime-elements/pull/2802/commits/05f80575ff2eec6620e0ea5ad1a9f7e5d0af1305